unblob/pool.py

```diff
@@ -72,11 +83,30 @@ def __init__(
             for _ in range(process_num)
         ]
         self._tid = threading.get_native_id()
+        self._signals_fired = []
+
+    def _handle_signal(self, signum: int, frame):
+        logger.debug(f"Signal ({signal.Signals(signum).name}) received")
+        self._signals_fired.append((signum, frame))
+        raise ShutDownRequired
+
+    def _register_signal_handler(self, sig: signal.Signals):
```

Comment:
Again a pointless abstraction. Just call `signal.signal()` directly where you would call this method.